File: calculator/src/main/java/com/turlir/calculator/interpreter/PolishInterpreter.java
package com.turlir.calculator.interpreter;


import java.math.BigDecimal;
import java.util.ArrayDeque;
import java.util.Deque;

/**
 * Очередь промежуточных результатов {@link BigDecimal} для вычисления значения математического выражения
 */
public class PolishInterpreter implements NotationInterpreter {

    private final Deque<BigDecimal> mQueue;

    public PolishInterpreter() {
        mQueue = new ArrayDeque<>();
    }

    @Override
    public BigDecimal poolDigit() {
        return mQueue.pop();
    }

    @Override
    public void pushDigit(BigDecimal value) {
        mQueue.push(value);
    }

}

File: calculator/src/main/kotlin/com/turlir/calculator/interpreter/PolishInterpreter.kt
package com.turlir.calculator.interpreter


import java.math.BigDecimal
import java.util.*

/**
 * Очередь промежуточных результатов [BigDecimal] для вычисления значения математического выражения
 */
class PolishInterpreter : NotationInterpreter {

    private val mQueue = ArrayDeque<BigDecimal>()

    override fun poolDigit() = mQueue.pop()!!

    override fun pushDigit(value: BigDecimal) {
        mQueue.push(value)
    }

}
